autohooks/utils.py

```diff
@@ -49,7 +51,7 @@ def exec_git(*args: str, ignore_errors: bool = False) -> str:
         exec_git("commit", "-m", "A new commit")
     """
     try:
-        cmd_args = ["git"]
+        cmd_args: List[Union[PathLike, str]] = ["git"]
```

Comment:
Not sure if this is correct. Shouldn't it be `List[str]` only? If `Union[PathLike, str]` is valid could you check if there is a test case and if not could you add one?

Reply:
i will check these changes again and come back to you. I think some of these changes where made in our joint coding session to eliminate other errors.

Reply:
`utils.py` cmd_args is typed as List[Union[PathLike, str]] because in the next line cmd_args.extend(args) is called. If you activate mypy and observe correct typing, the extend-method only accepts parameters of an type that is already in the List. And since an PathLike-Object is given as parameter for *args in line 69 (in get_git_directory_path()) cmd_args has to be prepared with the correct typing. 
`def extend(self, __iterable: Iterable[_T]) -> None: ...`
I realise that this typing is pretty confusing. So maybe an easier way is simply turning the PathLike-Object into an string early on and remove any PathLike parameters. I'm a bit unsure about properties that are lost in this process. I will check in what way this is covered in the test cases.

`terminal.py` Since the code doesn't prohibit the usage of terminal-functions before set_terminal is called i believe the only other typing solution would be to set some flags for mypy to ignore the lines / errors. Another option would be to define the behavior for each time "_term" is used when it is still "None". And of course i can update _set_terminal if we stay with this typing solution. Let me know what you would prefer (or if i missed anything)

Reply:
> I realise that this typing is pretty confusing. So maybe an easier way is simply turning the PathLike-Object into an string early on and remove any PathLike parameters. I'm a bit unsure about properties that are lost in this process. I will check in what way this is covered in the test cases.

In `test_utils.py` the PathLike-Object seems to get tested in exactly that way. So i think this change is convenient and might be even better tested
```
    def test_with_env_pwd(self):
        with TemporaryDirectory() as f:
            temp_path = Path(f)

            exec_git("-C", str(temp_path), "init")

            os.chdir(str(temp_path))

            git_dir_path = (temp_path / ".git").resolve()

            git_hook_dir_path = get_git_hook_directory_path()
            self.assertEqual(git_hook_dir_path, git_dir_path / "hooks")
```